src/main/jbake/content/blog/2017/03/17/apm-adjusting-sampling-rate.adoc

```diff
@@ -0,0 +1,385 @@
+default, using a concrete tracer (`APMTracer`) if the environment variable
+`HAWKULAR_APM_ENABLED` parses to `true`:
+
+`src/main/java/com/example/MainVerticle.java`
+```java
+public class MainVerticle extends AbstractVerticle {
+    private Tracer tracer = NoopTracerFactory.create();
+
+    @Override
+    public void start() {
+        boolean tracerEnabled = Boolean.parseBoolean(System.getenv("HAWKULAR_APM_ENABLED"));
+        if (tracerEnabled) {
+            tracer = new APMTracer();
+        }
+        vertx.createHttpServer()
+                .requestHandler((req) -> {
+                    Span span = tracer.buildSpan("hello-world-request").start();
+                    span.setTag("enabled", tracerEnabled);
+                    req.response().end(String.format("Hello World! Are we tracing this request? %s", System.getenv("HAWKULAR_APM_ENABLED")));
+                    span.finish();
+                })
+                .listen(8080);
+    }
+}
+```
+
+At this point, we are ready to test it! As we already have an OpenShift cluster
+running with our Hawkular APM server, we just need to run:
+
+```bash
+$ mvn clean fabric8:deploy
+```
+
+Once the build finishes running, we should see the following on OpenShift's Web
+Console:
+
+ifndef::env-github[]
+image::/img/blog/2017/2017-03-17-final-result.png[Final result]
+endif::[]
+ifdef::env-github[]
+image::../../../../../assets/img/blog/2017/2017-03-17-final-result.png[Final result]
+endif::[]
+
+To check whether our tracing is working, we can use a script like this:
+
+```bash
+$ export URL="http://conditional-tracer-myproject.192.168.2.107.xip.io"
+$ for i in `seq 1 10` ; do curl ${URL} 2>/dev/null | awk '{print $NF}' ; done | sort | uniq -c
+```
+
+An output similar to this one should be printed out:
+
+```
+8 false
+2 true
+```
+
+This means that 8 didn't have their traces published and two requests had
+their traces published to Hawkular APM. You can check that by logging into Hawkular
+APM (username: `admin`, password: `password`) and looking into the
+`Distributed Tracing` screen. There should be only two traces there.
+
+TIP: You might be tempted to try opening the URL in your browser and hitting
+refresh, but you'll notice that the OpenShift router tries to send you to the same
+backend service all the time. For this reason, it makes more sense to test on
+the command line with `curl`.
+
+== Wrapping up
+
+Deciding whether a transaction should be traced or not is a decision that should
+be outsourced as much as possible to the underlying Tracer, possibly with data
+coming from different sources. Having a simple switch on the code loading the
+Tracer, however, might prove invaluable to admins in determined situations.
+Given the simplicity of this solution, there's no reason not to have it!
+
+The full source code for this example can be found at
+link:https://github.com/hawkular/conditional-tracer[github.com/hawkular/conditional-tracer].
```

Comment:
Either this repository needs to be created, or I need to change the blog post to point to the repository at under my personal account: https://github.com/jpkrohling/conditional-tracer